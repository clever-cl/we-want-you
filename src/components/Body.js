import React, { useState } from 'react'
import "../styles/Body.css"
import Header from './Header'
import SongRow from './SongRow'
import SpotifyPlayer from 'react-spotify-player';
const Body = ({ spotify, response, data, getResponse }) => {
    const [id, setId] = useState(null)
    const size = {
        width: '100%',
        height: 300,
    };
//play song whe click
    const playSong = (id) => {
        setId(id)
    };
    return (
        <div className="body">
            <Header spotify={spotify} response={response} getResponse={getResponse} />
            <SpotifyPlayer
                uri={`spotify:track:${id}`}
                size={size}
            />
            <div className="body__songs">
                {data?.map(({album_image, album_name, artist_name, track_id}) => (
                    <div key={track_id}>
                    <SongRow album_image={album_image} album_name={album_name} artist_name={artist_name} track_id={track_id} playSong={playSong}/>
                    </div>
                ))}
            </div>
        </div>
    )
}

export default Body
