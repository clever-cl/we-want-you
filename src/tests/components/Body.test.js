import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import Body from '../../components/Body';

describe('Test in <Body />', () => {
  test('should show component correct', () => {
    const wrapper = shallow(<Body />)
    expect(wrapper).toMatchSnapshot()
  })


})