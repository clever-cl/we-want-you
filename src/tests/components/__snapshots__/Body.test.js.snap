// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Body /> should show component correct 1`] = `
<div
  className="body"
>
  <Header />
  <SpotifyPlayer
    size={
      Object {
        "height": 300,
        "width": "100%",
      }
    }
    theme="black"
    uri="spotify:track:null"
    view="list"
  />
  <div
    className="body__songs"
  />
</div>
`;
